src/staking/phase1/index.ts

```diff
@@ -0,0 +1,349 @@
+import { networks, Psbt, Transaction } from "bitcoinjs-lib";
+import { Phase1Params } from "../../types/params";
+import { UTXO } from "../../types/UTXO";
+import { StakingScripts } from "../stakingScript";
+import { StakingScriptData } from "../stakingScript";
+import { StakingError, StakingErrorCode } from "../../error";
+import { 
+  stakingTransaction, unbondingTransaction,
+  withdrawEarlyUnbondedTransaction,
+  withdrawTimelockUnbondedTransaction
+} from "../.";
+import { 
+  getPublicKeyNoCoord, isTaproot,
+  isValidBitcoinAddress, isValidNoCordPublicKey
+} from "../../utils/btc";
+import { getStakingTerm, validateStakingTxInputData } from "../../utils/staking";
+import { PsbtTransactionResult } from "../../types/transaction";
+
+interface StakerInfo {
+  address: string;
+  publicKeyHex: string;
+}
+
+interface Phase1StakingTransaction {
+  txHex: string;
+  outputIndex: number;
+  startHeight: number;
+  timelock: number;
+}
+
+interface Phase1Delegation {
+  stakingTxHashHex: string;
+  stakerPkHex: string;
+  finalityProviderPkHex: string;
+  stakingTx: Phase1StakingTransaction;
+}
+
+/**
+ * Phase1Staking is a class that provides methods to create staking transactions
+ * for Phase 1 of the Babylon Staking protocol.
+ * 
+ * The class requires a network and staker information to create staking
+ * transactions.
+ * The staker information includes the staker's address and 
+ * public key(without coordinates).
+ */
+export class Phase1Staking {
+  private network: networks.Network;
+  private stakerInfo: StakerInfo;
+
+  constructor(network: networks.Network, stakerInfo: StakerInfo) {
+    if (!isValidStakerInfo(stakerInfo, network)) {
+      throw new StakingError(
+        StakingErrorCode.INVALID_INPUT, "Invalid staker info",
+      );
+    }
+    this.network = network;
+    this.stakerInfo = stakerInfo;
+  }
+
+  /**
+   * Create a staking transaction for Phase 1 of the Babylon Staking protocol.
+   * 
+   * @param {Phase1Params} params - The staking parameters for Phase 1.
+   * @param {number} stakingAmountSat - The amount to stake in satoshis.
+   * @param {number} stakingTimeBlocks - The time to stake in blocks.
+   * @param {string} finalityProviderPublicKey - The finality provider's public key.
+   * @param {UTXO[]} inputUTXOs - The UTXOs to use as inputs for the staking 
+   * transaction.
+   * @param {number} feeRate - The fee rate for the transaction in satoshis per byte.
+   * 
+   * @returns {PsbtTransactionResult} - An object containing the unsigned psbt and fee
+   */
+  public createStakingTransaction = (
+    params: Phase1Params,
+    stakingAmountSat: number,
+    stakingTimeBlocks: number,
+    finalityProviderPublicKey: string,
```

Comment:
Typically we use `pk` instead of `PublicKey`. Also, does this contain a coordinate?